src/main/kotlin/com/zepben/evolve/services/network/EquivalentNetworkUtils.kt

```diff
@@ -0,0 +1,477 @@
+    ): EquivalentNetworkConnection {
+        val edgeEquipment = edgeNode.terminals.first().conductingEquipment
+            ?: throw IllegalStateException(
+                "[Internal Error]: ${edgeNode.typeNameAndMRID()} for EquivalentBranch must have a ConductingEquipment on " +
+                    "its terminal ${edgeNode.terminals.first().typeNameAndMRID()}"
+            )
+        //NOTE: This exception should never occur as an edgeNode by definition must have an associated ConductingEquipment with multiple equipment containers
+
+        val containersForEquivalentNetwork = getContainersForEdgeConnectivityNode(edgeNode)
+        val branchToEquipment = createEquivalentBranches(edgeEquipment via edgeNode).map { (phaseCode, equivalentBranch) ->
+            equivalentBranch.apply {
+                network.add(this)
+                containersForEquivalentNetwork.forEach {
+                    it.addEquipment(this)
+                    addContainer(it)
+                }
+
+                Terminal("$mRID-t${numTerminals() + 1}").also {
+                    network.add(it)
+                    addTerminal(it)
+                    it.conductingEquipment = this
+                    it.phases = phaseCode
+                    network.connect(it, edgeNode.mRID)
+                }
+            } to
+                createEquivalentEquipment(edgeEquipment via edgeNode via equivalentBranch).map { (phaseCode, conEq) ->
+                    conEq.apply {
+                        network.tryAdd(this)
+                        containersForEquivalentNetwork.forEach {
+                            it.addEquipment(this)
+                            this.addContainer(it)
+                        }
+
+                        val eqBranchTerminal = Terminal("${equivalentBranch.mRID}-t${equivalentBranch.numTerminals() + 1}").also {
+                            network.add(it)
+                            equivalentBranch.addTerminal(it)
+                            it.conductingEquipment = equivalentBranch
+                            it.phases = phaseCode
+                        }
+                        val conEqTerminal = Terminal("$mRID-t${numTerminals() + 1}").also {
+                            network.add(it)
+                            conEq.addTerminal(it)
+                            it.conductingEquipment = this
+                            it.phases = phaseCode
+                        }
+                        network.connect(eqBranchTerminal, conEqTerminal)
+                    }
+                }.toSet()
+        }.toMap()
+
+        return EquivalentNetworkConnection(edgeEquipment, edgeNode, branchToEquipment)
+    }
+
+    private fun getContainersForEdgeConnectivityNode(cn: ConnectivityNode): Set<EquipmentContainer> =
+        (cn.terminals.first().conductingEquipment?.containers?.toSet() ?: emptySet()).minus(
+            (cn.terminals as List).first().otherTerminals()
```

Comment:
You've still got a seemingly unnecessary `as List` here. Can you also add a bit of doc explaining what this does as well. I think it's almost self-explanatory from the signature, but we probably want to at least mention why this excludes `Site`s. You might also want to add a mention of Site's not counting in the docs at the top of the file?

Reply:
Done